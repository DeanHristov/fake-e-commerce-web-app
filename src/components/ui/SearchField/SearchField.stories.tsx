import type { Meta, StoryObj } from '@storybook/react';
import SearchField, { ISearchFieldProps } from './index';
import { MagnifyingGlassIcon } from '@heroicons/react/24/solid';

const meta: Meta<ISearchFieldProps> = {
  title: 'UI/SearchField',
  component: SearchField,
  argTypes: {
    onChange: { action: 'deBounce():' },
  },
  parameters: {
    layout: 'centered',
  },
};

type Story = StoryObj<ISearchFieldProps>;

export const Primary: Story = {
  args: {
    placeholder: 'Search here...',
    rightIcon: <MagnifyingGlassIcon className="w-7 h-7 text-zinc-100" />,
  },
};

export default meta;
