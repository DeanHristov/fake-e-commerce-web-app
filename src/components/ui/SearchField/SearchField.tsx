import { FC, useEffect, useState } from 'react';
import { useDebounce } from 'use-debounce';
import InputField, { IInputFieldProps } from '../InputField';

export interface ISearchFieldProps extends Omit<IInputFieldProps, 'onChange'> {
  delay?: number;
  onChange: (ev: string) => void;
}

const SearchField: FC<ISearchFieldProps> = ({
  placeholder,
  onChange,
  delay = 300,
  leftIcon,
  rightIcon,
  ...rest
}) => {
  const [valueText, setValueText] = useState<string>('');
  const [lazyValue] = useDebounce<string>(valueText, delay);

  useEffect(() => {
    if (lazyValue.length > 0) onChange(lazyValue);
  }, [lazyValue]);

  return (
    <InputField
      type="text"
      value={valueText}
      onChange={(ev) => setValueText(ev.target.value)}
      leftIcon={leftIcon}
      rightIcon={rightIcon}
      placeholder={placeholder}
    />
  );
};

export default SearchField;
