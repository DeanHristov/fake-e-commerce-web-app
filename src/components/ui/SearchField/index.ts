import SearchField, { ISearchFieldProps } from './SearchField';

export type { ISearchFieldProps };

export default SearchField;
