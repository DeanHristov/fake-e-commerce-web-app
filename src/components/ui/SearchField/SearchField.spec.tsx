import { render } from '@testing-library/react';
import SearchField from './SearchField';

describe('UI/Component <InputField {...} />', () => {
  it('Should render a default state of the Component', () => {
    const { container } = render(
      <SearchField onChange={() => {}} placeholder={'Text me...'} />,
    );

    expect(container).toMatchSnapshot();
  });
});
