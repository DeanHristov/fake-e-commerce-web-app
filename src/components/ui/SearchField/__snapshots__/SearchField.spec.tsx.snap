// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Component <InputField {...} /> Should render a default state of the Component 1`] = `
<div>
  <div
    class="relative w-full rounded-md h-full border border-gray-500"
  >
    <input
      class="h-full px-6 w-full rounded-lg bg-white text-slate-700 ease-in duration-200 focus:bg-white"
      placeholder="Text me..."
      type="text"
      value=""
    />
  </div>
</div>
`;
